drivers/sensor/mma8451q/mma8451q.c

```diff
@@ -0,0 +1,191 @@
+	 * - 8g mode (range = 2) has 12 fractional bits
+	 */
+	frac_bits = 14 - range;
+
+	/* Convert units to micro m/s^2. Intermediate results before the shift
+	 * are 40 bits wide.
+	 */
+	micro_ms2 = (raw * SENSOR_G) >> frac_bits;
+
+	/* The maximum possible value is 8g, which in units of micro m/s^2
+	 * always fits into 32-bits. Cast down to s32_t so we can use a
+	 * faster divide.
+	 */
+	val->val1 = (s32_t) micro_ms2 / 1000000;
+	val->val2 = (s32_t) micro_ms2 % 1000000;
+}
+
+static int mma8451q_channel_get(struct device *dev,
+				   enum sensor_channel chan,
+				   struct sensor_value *val)
+{
+	const struct mma8451q_config *config = dev->config->config_info;
+	struct mma8451q_data *data = dev->driver_data;
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+		mma8451q_accel_convert(val, data->x, config->range);
+		break;
+
+	case SENSOR_CHAN_ACCEL_Y:
+		mma8451q_accel_convert(val, data->y, config->range);
+		break;
+
+	case SENSOR_CHAN_ACCEL_Z:
+		mma8451q_accel_convert(val, data->z, config->range);
+		break;
+
+	case SENSOR_CHAN_ACCEL_XYZ:
+		mma8451q_accel_convert(val++, data->x, config->range);
+		mma8451q_accel_convert(val++, data->y, config->range);
+		mma8451q_accel_convert(val++, data->z, config->range);
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api mma8451q_driver_api = {
+	.sample_fetch = mma8451q_sample_fetch,
+	.channel_get = mma8451q_channel_get,
+};
+
+int mma8451q_init(struct device *dev)
+{
+	const struct mma8451q_config *config = dev->config->config_info;
+	struct mma8451q_data *data = dev->driver_data;
+
+	data->i2c = device_get_binding(config->i2c_name);
+	if (data->i2c == NULL) {
+		SYS_LOG_ERR("Failed to get pointer to %s device!",
+				config->i2c_name);
+		return -EINVAL;
+	}
+
+	u8_t whoami;
+
+	if (i2c_reg_read_byte(data->i2c, config->i2c_address,
+			      MMA8451Q_REG_WHOAMI, &whoami)) {
+		SYS_LOG_ERR("Could not get WHOAMI value");
+		return -EIO;
+	}
+
+	if (whoami != config->whoami) {
+		SYS_LOG_ERR("WHOAMI value received 0x%x, expected 0x%x",
+			    whoami, MMA8451Q_REG_WHOAMI);
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, config->i2c_address,
+				   MMA8451Q_CTRL_REG1, 0) < 0) {
+		SYS_LOG_ERR("Could not set accel in config mode");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, config->i2c_address,
+				   MMA8451Q_XYZ_DATA_CFG, config->range) < 0) {
+		SYS_LOG_ERR("Could not set range");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, config->i2c_address,
+				   MMA8451Q_CTRL_REG2, 0) < 0) {
+		SYS_LOG_ERR("Could not set to normal mode");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, config->i2c_address,
+				   MMA8451Q_CTRL_REG3, 0) < 0) {
+		SYS_LOG_ERR("Could not set to low polarity, push-pull output");
+		return -EIO;
+	}
+
+	if (i2c_reg_write_byte(data->i2c, config->i2c_address,
+				   MMA8451Q_CTRL_REG4, 1) < 0) {
+		SYS_LOG_ERR("Could not enable interrupt for data ready");
```

Comment:
I still don't understand why you're enabling the interrupt but not using it. You haven't set up any gpio callbacks. For this initial PR, I suggest removing this I2C transaction and the previous one which is also interrupt related, then adding interrupt support in a subsequent PR.

Reply:
ok - I'll cut it down to the basics